components/voting.tsx

```diff
@@ -117,6 +131,37 @@ export default class Voting extends React.PureComponent<VotingProps, VotingState
     return []
   }
 
+  async submit() {
+    const vote = {
+      Id: this.props.voteId,
+      Indices: this.state.votes.map(
+        id => this.props.sessions.indexOf(this.props.sessions.filter(s => s.Id === id)[0]) + 1,
+      ),
+      SessionIds: this.state.votes,
+      TicketNumber: this.state.ticketNumber,
+      VoterSessionId: getSessionId(),
+      VotingStartTime: this.props.startTime,
+    }
+    const headers: any = { 'Content-Type': 'application/json' }
+    this.setState({ submitInProgress: true, submitError: false })
+    try {
+      const response = await fetch(this.props.submitVoteUrl, {
+        body: JSON.stringify(vote),
+        headers,
+        method: 'POST',
+      })
+
+      if (!response.ok) {
+        this.setState({ submitInProgress: false, submitError: true })
+      } else {
+        this.setState({ submitInProgress: false, submitted: true })
+        this.writeToStorage('ddd-voting-submitted', 'true')
```

Comment:
we should write to local storage in the setState callback to ensure consistency between app and persisted state.